### PR TITLE
Improve the documentation

### LINKED ISSUE #27: What is 'q' in inputs?
I would like to add this to the README.md.

What is `q` in `inputs(hypes, q, phase, data_dir)`?

### PR DESCRIPTION
**Pull request by MartinThoma**
- Extend documentation for inputs()
- Fix requirements to make it work with readthedocs.org
- Add explanation of q (fixes #27)
- Ignore non-project related stuff
- Expand docstrings and apply Numpydoc format

### CODE REVIEW

File: bin/tv-train [hunk 1]
```diff
@@ -179,7 +189,18 @@ def do_eval(hypes, eval_correct, phase, sess):
 
 
 def run_training(hypes):
-    """Train model for a number of steps."""
+    """
+    Train model for a number of steps.
+
+    This traines the model for at most hypes['solver']['max_steps'].
+
+    TODO: What is stored where?
```
Comment (MartinThoma) [inline bin/tv-train:197]: @MarvinTeichmann

File: tensorvision/eval.py [hunk 2]
```diff
@@ -36,15 +36,16 @@
 
 def evaluate(train_dir):
   """
-  Loads the model and runs evaluation
+  Load the model and run evaluation.
+
+  TODO: Which kind of evaluation? Does it simply get printed to standard
```
Comment (MartinThoma) [inline tensorvision/eval.py:41]: @MarvinTeichmann

**Comment by MartinThoma** [comment]
See http://tensorvision.readthedocs.io/en/latest/ for the generated documentation.

**Comment by MartinThoma** [comment]
@MarvinTeichmann What do you think about rewriting the README.md to be more like the one of https://github.com/Lasagne/Lasagne?

I would like to have their structure "what is it", "how do I install it", "where is more documentation"

I think I would move the configuration stuff rather to readthedocs.org, to make the README less cluttered and also to prevent having two documentations.

**Comment by MarvinTeichmann** [comment]
The documentation looks great! Go ahead and merge, if you feel so.

> I would like to have their structure "what is it", "how do I install it", "where is more documentation"

Yes, I like this. Sounds like a very good structure.

**Comment by MartinThoma** [comment]
@MarvinTeichmann Ok, I moved the stuff from the README to the real documentation. Could you please answer my TODOs (e.g. http://tensorvision.readthedocs.io/en/latest/modules/utils.html), so that we fix those with this pull request? (See more above, just search this page for "TODO" and add a line note)

**Comment by MarvinTeichmann** [comment]
Ok, I will try to get some time for it later today. I guess it will be easiest, if I fix the TODOs directly in a commit?

**Comment by MartinThoma** [comment]
Sure, go ahed. I will not force-push the documentation branch until you fixed those.

**Comment by MartinThoma** [comment]
@MarvinTeichmann Could you please fix the TODOs here? I don't want to have merge conflicts just because of a few docstrings.

**Comment by MarvinTeichmann** [comment]
@MartinThoma It might be a better Idea to fix the doc strings in #32. There has been quite a lot of restructuring made. 

Additionally I am not sure, whether it is a great Idea to maintain a long running feature Branch for Documentation. I think a smooth development process is more important than a completely clean commit history. The development should not be kept back by out branch structure.

**Comment by MartinThoma** [comment]
@MarvinTeichmann So should I merge now?

**Pull request merged**